pkg/errors/sdk_errors.go

```diff
@@ -81,9 +102,18 @@ func CheckError(resp *http.Response, expectedStatusCodes ...int) SDKError {
 	var content map[string]interface{}
 	_ = json.Unmarshal(body, &content)
 
-	if msg, ok := content[errorKey]; ok {
-		if v, ok := msg.(string); ok {
-			return NewSDKErrorWithStatus(New(v), resp.StatusCode)
+	if msg, ok := content[messageKey]; ok {
```

Comment:
Since we are expecting a known format of the error, we should simplify this by unmarshalling the body to the expected struct  (rather than `map[string]interface{}`) and handling unmarshal errors, rather than using this unnecessarily complex approach.